test/Microsoft.AspNet.Antiforgery.Test/DefaultAntiforgeryTest.cs

```diff
@@ -313,11 +313,30 @@ public void GetTokens_ExistingValidCookieToken_GeneratesANewFormToken()
             var tokenset = antiforgery.GetTokens(context.HttpContext);
 
             // Assert
-            Assert.Null(tokenset.CookieToken);
+            Assert.NotNull(tokenset.CookieToken);
+            Assert.Equal("serialized-old-cookie-token", tokenset.CookieToken);
             Assert.Equal("serialized-form-token", tokenset.FormToken);
         }
 
         [Fact]
+        public void GetAndStoreTokens_ExistingValidCookieToken_SaveOnce()
+        {
+            var context = CreateMockContext(
+                new AntiforgeryOptions(),
+                useOldCookie: true,
+                isOldCookieValid: true);
+            var antiforgery = GetAntiforgery(context);
+
+            var tokenSet = antiforgery.GetAndStoreTokens(context.HttpContext);
+            
+            //Strict mock would have complained if SaveCookieToken was called.
+
+            Assert.NotNull(tokenSet.CookieToken);
+            Assert.Equal("serialized-old-cookie-token", tokenSet.CookieToken);
+            Assert.Equal("serialized-form-token", tokenSet.FormToken);
+        }
```

Comment:
The above test was intended to be the test of the property. If the CookieToken is not new, then we shouldn't be trying to save it again. I've added a test to ensure we save it when we actually do create a new token as well, will put that up with the update.